documentation/implementation_events.rst

```diff
@@ -0,0 +1,85 @@
+Event Implementation
+=======================
+
+This document provides guidance an rational on the implementation of events in
+AMICI. Events include any discontinuities in the right hand side of the
+differential equation. There are three types of discontinuities:
+
+- **Solution Jump Discontinuities** can be created by SBML  events or delta
+  functions in the right hand side.
+
+- **Right-Hand-Side Jump Discontinuities** result in removable
+  discontinuities in the solution and can be created by SBML Piecewise or
+  Heaviside functions.
+
+- **Right-Hand-Side Removeable Discontinuities** do not lead to
+  discontinuities in the solution, but may lead to discontinuous higher
+  order temporal derivatives and can be created by functions such as max or
+  min.
+
+Mathematical Considerations
+---------------------------
+
+A detailed mathematical description of the required sensitivity formulas is
+provided in
+
+* Fröhlich, F., Theis, F. J., Rädler, J. O., & Hasenauer, J. (2017).
+  Parameter estimation for dynamical systems with discrete events and logical
+  operations. Bioinformatics, 33(7), 1049-1056.
+  doi:`10.1093/bioinformatics/btw764 <https://doi.org/10.1093/bioinformatics/btw764>`_.
+
+Algorithmic Considerations
+--------------------------
+
+Right-Hand-Side Jump Discontinuities
```

Comment:
Order of subsections in "Algorithmic Considerations" is different to the order of items in the bold list above.